### PR TITLE
CLOUDP-225849: Add scaffolding of the openapi cli

### PR DESCRIPTION
**Pull request by andreaangiolillo**
Ticket: [CLOUDP-225849](https://jira.mongodb.org/browse/CLOUDP-225849)

# Description
This PR adds the scaffolding of the openapi cli that we will use to merge OASes. In particular, this PR defines the cli builder and the `merge` command without any logic. 

```bash
make build                                                                                                                                          4s  1.21.4  16.18.0
==> Building openapicli binary
go build -ldflags "-s -w -X github.com/mongodb/openapi/tools/cli/internal/version.GitCommit=e5fb49fd94796ca0a81618431d81c38d6ff45f7f -X github.com/mongodb/openapi/tools/cli/internal/version.Version=v0.0.1" -o ./bin/openapicli ./cmd

./bin/openapicli -v                                                                                                                                  1.21.4  16.18.0
openapicli version: next
git version: next
Go version: go1.21.4
   os: darwin
   arch: arm64
   compiler: gc
```

# Next steps
- Open a follow up PR to add the logic of the merge command
- Open a follow up PR to add tests

### CODE REVIEW

File: tools/cli/go.mod [hunk 1]
```diff
@@ -0,0 +1,23 @@
+module mongodb/openapi/tools/cli
+
+go 1.21.4
```
Comment (gssbzn) [inline tools/cli/go.mod:3]:
```suggestion
go 1.22
```

File: tools/cli/bin/openapicli
Comment (gssbzn) [inline tools/cli/bin/openapicli:1]: should be ignored

File: tools/cli/Makefile [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	go mod download
+	go mod tidy
+
+.PHONY: setup
+setup: deps ## Set up dev env
+
+.PHONY: fmt
+fmt: ### Format all go files with goimports and gofmt
+	find . -name "*.go" -not -path "./vendor/*" -not -path "./internal/mocks" -exec gofmt -w "{}" \;
+	find . -name "*.go" -not -path "./vendor/*" -not -path "./internal/mocks" -exec goimports -l -w "{}" \;
```
Comment (gssbzn) [inline tools/cli/Makefile:29]: I don't believe we have vendor or mocks

File: tools/cli/bin/openapicli
Comment (andreaangiolillo) [reply tools/cli/bin/openapicli:1]: ops

File: tools/cli/internal/cli/merge/merge.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+func (o *Opts) PreRunE(args []string) error {
+	// To Add in follow up PR: CLOUDP-225849
+	return nil
+}
+
+func Builder() *cobra.Command {
+	opts := &Opts{}
+
+	cmd := &cobra.Command{
+		Use:   "merge [base-spec] [spec-1] [spec-2] [spec-3] ... [spec-n]",
```
Comment (gssbzn) [inline tools/cli/internal/cli/merge/merge.go:57]: let;s use flags, they are friendlier

File: tools/cli/internal/cli/merge/merge.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+	return nil, nil
+
+}
+func (o *Opts) saveFile(data []byte) error {
+	if err := os.WriteFile(o.outputPath, data, 0644); err != nil {
+		return err
+	}
+	_, _ = fmt.Printf("\nFederated Spec was saved in '%s'\n", o.outputPath)
+	return nil
+}
```
Comment (gssbzn) [inline tools/cli/internal/cli/merge/merge.go:46]: don't see this used but let's default to stdout first and then add a `--out` flag for the file

Comment (andreaangiolillo) [reply tools/cli/internal/cli/merge/merge.go]: removed the logic for now, I will open a follow-up PR to add the logic and address this comment. Thanks!

File: tools/cli/internal/cli/root/openapi/builder.go [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+   os: %s
+   arch: %s
+   compiler: %s
+`
+)
+
+// Builder conditionally adds children commands as needed.
+// This is important in particular for Atlas as it dynamically sets flags for cluster creation and
+// this can be slow to timeout on environments with limited internet access (Ops Manager).
+func Builder(argsWithoutProg []string) *cobra.Command {
```
Comment (gssbzn) [inline tools/cli/internal/cli/root/openapi/builder.go:40]: unused? maybe a good idea to add golangci-lint

File: tools/cli/internal/cli/merge/merge.go (hunk 3 above)
Comment (andreaangiolillo) [reply tools/cli/internal/cli/merge/merge.go:50]: I will add the logic in a follow-up PR. Out of curiosity, how do you envision the usage of the command when we want to merge multiple spec into the base one? Based on your comment, I was thinking about moving the base spec to a flag and leaving the other as args. What do you think?

File: tools/cli/internal/cli/root/openapi/builder.go (hunk 5 above)
Comment (andreaangiolillo) [reply tools/cli/internal/cli/root/openapi/builder.go:40]: this is used in `openapicli.go`. Yes, I will add golanci-lint in a follow up PR

Comment (gssbzn) [reply tools/cli/internal/cli/root/openapi/builder.go:40]: either same as you have here first is base or we can have two flags `--base` `--other...` and why flags are usually more friendly

File: tools/cli/internal/cli/merge/merge.go (hunk 3 above)
Comment (gssbzn) [reply tools/cli/internal/cli/merge/merge.go:50]: either same as you have here, first is base or `--base` `--external...` or something like that and why flags tend to be friendlier

File: tools/cli/internal/cli/root/openapi/builder.go (hunk 5 above)
Comment (gssbzn) [reply tools/cli/internal/cli/root/openapi/builder.go:40]: but I don't see it used, this feels like a copy/past from atlas cli where we used to know if to enforce auth or not

Comment (gssbzn) [reply tools/cli/internal/cli/root/openapi/builder.go:40]: sorry to be clear the unused is `argsWithoutProg`

Comment (andreaangiolillo) [reply tools/cli/internal/cli/root/openapi/builder.go:40]: ah sorry, yes I will remove it

**Review by gssbzn — APPROVED** [review]
:shipit:

**Pull request merged**